app/services/licences/supplementary/process-imported-licence.service.js

```diff
@@ -0,0 +1,125 @@
+'use strict'
+
+/**
+ * Processes a licence that has been imported with at least one changed 'end' date (expired, lapsed, or revoked)
+ * @module ProcessImportedLicenceService
+ */
+
+const DetermineBillingYearsService = require('./determine-billing-years.service.js')
+const FetchExistingLicenceDetailsService = require('./fetch-existing-licence-details.service.js')
+const { determineCurrentFinancialYear } = require('../../../lib/general.lib.js')
+const PersistSupplementaryBillingFlagsService = require('./persist-supplementary-billing-flags.service.js')
+
+const SROC_START_DATE = new Date('2022-04-01')
+
+/**
+ * Processes a licence that has been imported with at least one changed 'end' date (expired, lapsed, or revoked)
+ *
+ * @param {object} importedLicence - Object representing the updated licence being imported
+ * @param {string} licenceId - The UUID of the licence being imported
+ *
+ * @returns {Promise} A promise is returned but it does not resolve to anything we expect the caller to use
+ */
+async function go (importedLicence, licenceId) {
+  const existingLicenceDetails = await FetchExistingLicenceDetailsService.go(licenceId)
+  const earliestChangedDate = _earliestChangedDate(importedLicence, existingLicenceDetails)
+
+  // If not set it means none of the dates changed were before the current financial year end so there is no reason
+  // to change anything on the flags
+  if (!earliestChangedDate) {
+    return
+  }
+
+  const flagForPreSrocSupplementary = _flagForPresrocSupplementary(existingLicenceDetails, earliestChangedDate)
+  const flagForSrocSupplementary = _flagForSrocSupplementary(existingLicenceDetails)
+  const twoPartTariffBillingYears = _flagForTwoPartTariffSupplementary(existingLicenceDetails, earliestChangedDate)
+
+  await PersistSupplementaryBillingFlagsService.go(
```

Comment:
We can just return this.

```suggestion
  return PersistSupplementaryBillingFlagsService.go(
```